lib/shared/src/models/sync.test.ts

```diff
@@ -658,7 +658,7 @@ describe('maybeAdjustContextWindows', () => {
             },
         ] satisfies ServerModel[]
 
-        const result = maybeAdjustContextWindows(models, false, true, true)
+        const result = maybeAdjustContextWindows(models, false, false, true)
```

Comment:
```suggestion
        const result = maybeAdjustContextWindows(models, {
            tier: 'free',
            longContextWindowFlagEnabled: true,
        })
```

This is hard to follow and read. We could use an object here to make it more readable.